# Goal Behaviour Resolution: Transfer Response Model

Consequent Reversibility was not included in the final escape models because, in this subgroup structure, it is partially encoded by Goal_Type and produced unstable estimates when entered together with Goal_Type.

Import Libraries

In [45]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

## Overall Subgroup
This subset of data explores observations when _Focus_ = I and _Focus_= They.
1. Create resolved_transfer
2. Counts and Proportions for Goals vs. no_goals
3. Logistic Models
    - RC1: Agency (goal_frequent vs. goal_non_frequent)
    - RC2: Goal Type (goal_frequent vs. goal_non_frequent)
    - RC3: Additive Model (goal_frequent vs. goal_non_frequent)
    - RC4: Interactive Model (goal_frequent vs. goal_non_frequent)
    - RC5: Agency  (no_goals)
4. Model Comparison
5. Summary Table

Read Data

In [46]:
subgroup_theoretical = pd.read_csv("../../data/processed/subgroup_theoretical.csv")

1. Create resolved_correct

Binary Outcome: include correct responses from escape_L2 observations.

In [47]:
subgroup_theoretical["escape_L2"] = (
    subgroup_theoretical["Response_Full"] != "L2_other"
).astype(int)

In [48]:
escapees = subgroup_theoretical[subgroup_theoretical["escape_L2"] == 1].copy()

In [49]:
escapees["resolved_transfer"] = (escapees["Response_Full"] == "L1_transfer").astype(int)

In [50]:
goals_escapees = escapees[escapees["Goal_Type"] != "no_goal"].copy()

In [51]:
no_goal_escapees = escapees[escapees["Goal_Type"] == "no_goal"].copy()

2. Counts and Proportions

In [52]:
resolved_transfer_counts = pd.crosstab(
    escapees["resolved_transfer"],
    escapees["Response_Full"])

resolved_transfer_counts

Response_Full,L1_transfer,correct,missing_response
resolved_transfer,,,
0,0,61,3
1,96,0,0


In [53]:
goals_resolved_transfer_counts = pd.crosstab(
    goals_escapees["resolved_transfer"],
    goals_escapees["Response_Full"])

goals_resolved_transfer_counts

Response_Full,L1_transfer,correct,missing_response
resolved_transfer,,,
0,0,43,1
1,62,0,0


In [54]:
no_goal_resolved_transfer_counts = pd.crosstab(
    no_goal_escapees["resolved_transfer"],
    no_goal_escapees["Response_Full"])

no_goal_resolved_transfer_counts

Response_Full,L1_transfer,correct,missing_response
resolved_transfer,,,
0,0,18,2
1,34,0,0


In [55]:
resolved_transfer_props = escapees["resolved_transfer"].value_counts(normalize = True)
resolved_transfer_props

resolved_transfer
1    0.6
0    0.4
Name: proportion, dtype: float64

From 160 observations, almost 38% of responses that had escaped *L2_other* options were resolved correctly. 

Counts and Proportions by Condition:

In [56]:
counts_condition = pd.crosstab(
    [escapees["Goal_Type"], escapees["Agent"]],
    escapees["resolved_transfer"],
    margins = True)

counts_condition

resolved_transfer         0   1  All
Goal_Type         Agent             
goal_frequent     0      15  13   28
                  1      10  12   22
goal_non_frequent 0      12  21   33
                  1       7  16   23
no_goal           0      11  23   34
                  1       9  11   20
All                      64  96  160

In [57]:
goals_counts_condition = pd.crosstab(
    [goals_escapees["Goal_Type"], goals_escapees["Agent"]],
    goals_escapees["resolved_transfer"],
    margins = True)

goals_counts_condition

resolved_transfer         0   1  All
Goal_Type         Agent             
goal_frequent     0      15  13   28
                  1      10  12   22
goal_non_frequent 0      12  21   33
                  1       7  16   23
All                      44  62  106

In [58]:
no_goal_counts_condition = pd.crosstab(
    [no_goal_escapees["Goal_Type"], no_goal_escapees["Agent"]],
    no_goal_escapees["resolved_transfer"],
    margins = True)

no_goal_counts_condition

resolved_transfer   0   1  All
Goal_Type Agent               
no_goal   0        11  23   34
          1         9  11   20
All                20  34   54

In [59]:
goals_props_condition = pd.crosstab(
    [goals_escapees["Goal_Type"], goals_escapees["Agent"]],
    goals_escapees["resolved_transfer"],
    normalize= "index")

goals_props_condition

resolved_transfer               0         1
Goal_Type         Agent                    
goal_frequent     0      0.535714  0.464286
                  1      0.454545  0.545455
goal_non_frequent 0      0.363636  0.636364
                  1      0.304348  0.695652

In [60]:
no_goal_props_condition = pd.crosstab(
    [no_goal_escapees["Goal_Type"], no_goal_escapees["Agent"]],
    no_goal_escapees["resolved_transfer"],
    normalize= "index")

no_goal_props_condition

resolved_transfer         0         1
Goal_Type Agent                      
no_goal   0        0.323529  0.676471
          1        0.450000  0.550000

COMMENT GENERAL PROPORTIONS

3. Logistic Models

Import Libraries

In [61]:
import statsmodels.formula.api as smf
from scipy.stats import chi2

In [62]:
escapees["resolved_transfer"].value_counts()

resolved_transfer
1    96
0    64
Name: count, dtype: int64

In [63]:
goals_escapees["resolved_transfer"].value_counts()

resolved_transfer
1    62
0    44
Name: count, dtype: int64

In [64]:
no_goal_escapees["resolved_transfer"].value_counts()

resolved_transfer
1    34
0    20
Name: count, dtype: int64

**Model RT1**

resolved_transfer ~ Agent

In [65]:
RT1 = smf.logit(
    "resolved_transfer ~ Agent",
    data=goals_escapees
    ).fit()

print(RT1.summary())

Optimization terminated successfully.
         Current function value: 0.676537
         Iterations 4
                           Logit Regression Results                           
Dep. Variable:      resolved_transfer   No. Observations:                  106
Model:                          Logit   Df Residuals:                      104
Method:                           MLE   Df Model:                            1
Date:                Wed, 01 Jul 2026   Pseudo R-squ.:                0.003127
Time:                        14:12:02   Log-Likelihood:                -71.713
converged:                       True   LL-Null:                       -71.938
Covariance Type:            nonrobust   LLR p-value:                    0.5024
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      0.2305      0.258      0.894      0.371      -0.275       0.736
Agent          0.2685      0.

**Model RT2**

resolved_transfer ~ Goal_Type

In [66]:
RT2 = smf.logit(
    "resolved_transfer ~ Goal_Type",
    data= goals_escapees
    ).fit()

print(RT2.summary())

Optimization terminated successfully.
         Current function value: 0.665366
         Iterations 4
                           Logit Regression Results                           
Dep. Variable:      resolved_transfer   No. Observations:                  106
Model:                          Logit   Df Residuals:                      104
Method:                           MLE   Df Model:                            1
Date:                Wed, 01 Jul 2026   Pseudo R-squ.:                 0.01959
Time:                        14:12:02   Log-Likelihood:                -70.529
converged:                       True   LL-Null:                       -71.938
Covariance Type:            nonrobust   LLR p-value:                   0.09320
                                     coef    std err          z      P>|z|      [0.025      0.975]
--------------------------------------------------------------------------------------------------
Intercept                       6.516e-17      0.283    2.3e-16     

**Model RT3**

resolved_transfer ~ Agent + Goal_Type

In [67]:
RT3 = smf.logit(
    "resolved_transfer ~ Agent + Goal_Type",
    data= goals_escapees
    ).fit()

print(RT3.summary())

Optimization terminated successfully.
         Current function value: 0.662848
         Iterations 4
                           Logit Regression Results                           
Dep. Variable:      resolved_transfer   No. Observations:                  106
Model:                          Logit   Df Residuals:                      103
Method:                           MLE   Df Model:                            2
Date:                Wed, 01 Jul 2026   Pseudo R-squ.:                 0.02330
Time:                        14:12:02   Log-Likelihood:                -70.262
converged:                       True   LL-Null:                       -71.938
Covariance Type:            nonrobust   LLR p-value:                    0.1871
                                     coef    std err          z      P>|z|      [0.025      0.975]
--------------------------------------------------------------------------------------------------
Intercept                         -0.1305      0.335     -0.389     

**Model RT4**

resolved_transfer ~ Agent * Goal_Type

In [68]:
RT4 = smf.logit(
    "resolved_transfer ~ Agent * Goal_Type",
    data=goals_escapees
    ).fit()

print(RT4.summary())

Optimization terminated successfully.
         Current function value: 0.662824
         Iterations 5
                           Logit Regression Results                           
Dep. Variable:      resolved_transfer   No. Observations:                  106
Model:                          Logit   Df Residuals:                      102
Method:                           MLE   Df Model:                            3
Date:                Wed, 01 Jul 2026   Pseudo R-squ.:                 0.02333
Time:                        14:12:02   Log-Likelihood:                -70.259
converged:                       True   LL-Null:                       -71.938
Covariance Type:            nonrobust   LLR p-value:                    0.3398
                                           coef    std err          z      P>|z|      [0.025      0.975]
--------------------------------------------------------------------------------------------------------
Intercept                               -0.1431      0.3

**Model RT5 (no_goals only)**

resolved_transfer ~ Agent

In [69]:
RT5 = smf.logit(
    "resolved_transfer ~ Agent",
    data=no_goal_escapees
    ).fit()

print(RT5.summary())

Optimization terminated successfully.
         Current function value: 0.651219
         Iterations 4
                           Logit Regression Results                           
Dep. Variable:      resolved_transfer   No. Observations:                   54
Model:                          Logit   Df Residuals:                       52
Method:                           MLE   Df Model:                            1
Date:                Wed, 01 Jul 2026   Pseudo R-squ.:                 0.01204
Time:                        14:12:02   Log-Likelihood:                -35.166
converged:                       True   LL-Null:                       -35.594
Covariance Type:            nonrobust   LLR p-value:                    0.3546
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      0.7376      0.367      2.012      0.044       0.019       1.456
Agent         -0.5369      0.

4. Model Comparisons

In [70]:
from scipy.stats import chi2

def lr_test(model_small, model_large):

    LR = 2 * (model_large.llf - model_small.llf)

    df = model_large.df_model - model_small.df_model

    p = chi2.sf(LR, df)

    return pd.Series({
        "LR": LR,
        "df": df,
        "p_value": p
    })

In [71]:
lr_test(RT2, RT3)

LR         0.533748
df         1.000000
p_value    0.465036
dtype: float64

In [72]:
lr_test(RT2, RT4)

LR         0.538881
df         2.000000
p_value    0.763807
dtype: float64

5. Summary Table

model_name = ["RC1", "RC2", "RC2", "RC3", "RC3", "RC3", "RC4", "RC4", "RC4", "RC4", "RC4"]

outcome = ["resolved_correct","resolved_correct", "resolved_correct", "resolved_correct", "resolved_correct", "resolved_correct", "resolved_correct", "resolved_correct", "resolved_correct", "resolved_correct", "resolved_correct"]

subgroup = ["Overall", "Overall", "Overall", "Overall", "Overall", "Overall", "Overall", "Overall", "Overall", "Overall", "Overall"]

formula = ["resolved_correct ~ Agent", "resolved_correct ~ Goal_Type", "resolved_correct ~ Goal_Type", "resolved_correct ~ Agent + Goal_Type", "resolved_correct ~ Agent + Goal_Type", "resolved_correct ~ Agent + Goal_Type", "resolved_correct ~ Agent * Goal_Type", "resolved_correct ~ Agent * Goal_Type", "resolved_correct ~ Agent * Goal_Type", "resolved_correct ~ Agent * Goal_Type", "resolved_correct ~ Agent * Goal_Type"]

predictor= ["Agent", "goal_non_frequent", "no_goal", "goal_non_frequent", "no_goal", "Agent",  "goal_non_frequent", "no_goal", "Agent", "Agent:goal_non_frequent", "Agent:no_goal"]

coef = [1.1086, 0.5833, 0.3646, 0.6038, 0.3785, 1.1182, 0.6061, 0.7538, 1.7047, 0.1315, -1.5422]

coef_p = [0.013, 0.193, 0.396, 0.185, 0.387, 0.013, 0.224, 0.141, 0.035, 0.923, 0.143]

LLR_p = [0.007642, 0.4060, 0.4060, 0.02954, 0.02954, 0.02954, 0.03590, 0.03590, 0.03590, 0.03590, 0.03590]

N= [198, 198, 198, 198, 198, 198, 198, 198, 198, 198, 198]

Takeaway = []

correct_summary_overall = pd. DataFrame({
    "Model": model_name,
    "Outcome": outcome,
    "Subgroup": subgroup,
    "Formula": formula, 
    "Predictor": predictor,
    "Coef": coef,
    "Coef_p": coef_p,
    "Model_LLR_p": LLR_p,
    "N": N
})

correct_summary_overall

## Actors Subgroup
This subset of data explores observations only when _Focus_ = I.
1. Create resolved_correct for goals and no_goals
2. Counts and Proportions for goals and no_goals
3. Logistic Models
    - RC1_A: Agency (goal_frequent vs. goal_non_frequent)
    - RC2_A: Goal Type (goal_frequent vs. goal_non_frequent)
    - RC3_A: Additive Model (goal_frequent vs. goal_non_frequent)
    - RC4_A: Interactive Model (goal_frequent vs. goal_non_frequent)
    - RC5_A: Agency (no_goal)
4. Model Comparison
5. Summary Table

Read Data

In [73]:
subgroup_actors = subgroup_theoretical[subgroup_theoretical["Focus"] == "I"].copy()

1. Create resolved_correct

Binary Outcome: include correct responses from escape_L2 observations within actors.

In [74]:
subgroup_actors["escape_L2"] = (
    subgroup_actors["Response_Full"] != "L2_other"
).astype(int)

In [75]:
escapees_actors = subgroup_actors[subgroup_actors["escape_L2"] == 1].copy()

In [76]:
escapees_actors["resolved_transfer"] = (escapees_actors["Response_Full"] == "L1_transfer").astype(int)

In [77]:
goals_escapees_actors = escapees_actors[escapees_actors["Goal_Type"] != "no_goal"].copy()

In [78]:
no_goal_escapees_actors = escapees_actors[escapees_actors["Goal_Type"] == "no_goal"].copy()

2. Counts and Proportions

In [79]:
resolved_transfer_counts = pd.crosstab(
    escapees_actors["resolved_transfer"],
    escapees_actors["Response_Full"])

resolved_transfer_counts

Response_Full,L1_transfer,correct,missing_response
resolved_transfer,,,
0,0,40,3
1,53,0,0


In [80]:
goals_resolved_transfer_counts = pd.crosstab(
    goals_escapees_actors["resolved_transfer"],
    goals_escapees_actors["Response_Full"])

goals_resolved_transfer_counts

Response_Full,L1_transfer,correct,missing_response
resolved_transfer,,,
0,0,29,1
1,31,0,0


In [81]:
no_goal_resolved_transfer_counts = pd.crosstab(
    no_goal_escapees_actors["resolved_transfer"],
    no_goal_escapees_actors["Response_Full"])

no_goal_resolved_transfer_counts

Response_Full,L1_transfer,correct,missing_response
resolved_transfer,,,
0,0,11,2
1,22,0,0


In [82]:
resolved_transfer_props = escapees_actors["resolved_transfer"].value_counts(normalize = True)
resolved_transfer_props

resolved_transfer
1    0.552083
0    0.447917
Name: proportion, dtype: float64

From 96 actors' observations (from the original 160), almost 41% of responses that managed to escape *L2_other* options were resolved correctly.

In [83]:
goals_resolved_transfer_props = goals_escapees_actors["resolved_transfer"].value_counts(normalize = True)
goals_resolved_transfer_props

resolved_transfer
1    0.508197
0    0.491803
Name: proportion, dtype: float64

In [84]:
no_goal_resolved_transfer_props = no_goal_escapees_actors["resolved_transfer"].value_counts(normalize = True)
no_goal_resolved_transfer_props

resolved_transfer
1    0.628571
0    0.371429
Name: proportion, dtype: float64

Counts and Proportions by Condition:

In [85]:
counts_condition = pd.crosstab(
    [escapees_actors["Goal_Type"], escapees_actors["Agent"]],
    escapees_actors["resolved_transfer"],
    margins = True)

counts_condition

resolved_transfer         0   1  All
Goal_Type         Agent             
goal_frequent     0      10   6   16
                  1       7   7   14
goal_non_frequent 0       9   7   16
                  1       4  11   15
no_goal           0       8  13   21
                  1       5   9   14
All                      43  53   96

In [86]:
goals_counts_condition = pd.crosstab(
    [goals_escapees_actors["Goal_Type"], goals_escapees_actors["Agent"]],
    goals_escapees_actors["resolved_transfer"],
    margins = True)

goals_counts_condition

resolved_transfer         0   1  All
Goal_Type         Agent             
goal_frequent     0      10   6   16
                  1       7   7   14
goal_non_frequent 0       9   7   16
                  1       4  11   15
All                      30  31   61

In [87]:
no_goal_counts_condition = pd.crosstab(
    no_goal_escapees_actors["Agent"],
    no_goal_escapees_actors["resolved_transfer"],
    margins = True)

no_goal_counts_condition

resolved_transfer,0,1,All
Agent,,,
0,8,13,21
1,5,9,14
All,13,22,35


In [88]:
goals_props_condition = pd.crosstab(
    [goals_escapees_actors["Goal_Type"], goals_escapees_actors["Agent"]],
    goals_escapees_actors["resolved_transfer"],
    normalize= "index")

goals_props_condition

resolved_transfer               0         1
Goal_Type         Agent                    
goal_frequent     0      0.625000  0.375000
                  1      0.500000  0.500000
goal_non_frequent 0      0.562500  0.437500
                  1      0.266667  0.733333

In [89]:
no_goal_props_condition = pd.crosstab(
    no_goal_escapees_actors["Agent"],
    no_goal_escapees_actors["resolved_transfer"],
    normalize= "index")

no_goal_props_condition

resolved_transfer,0,1
Agent,,
0,0.380952,0.619048
1,0.357143,0.642857


3. Logistic Models

In [90]:
escapees_actors["resolved_transfer"].value_counts()

resolved_transfer
1    53
0    43
Name: count, dtype: int64

In [91]:
goals_escapees_actors["resolved_transfer"].value_counts()

resolved_transfer
1    31
0    30
Name: count, dtype: int64

In [92]:
no_goal_escapees_actors["resolved_transfer"].value_counts()

resolved_transfer
1    22
0    13
Name: count, dtype: int64

**RC1_A Model** 

resolved_correct ~ Agent

In [93]:
RT1_A = smf.logit(
    "resolved_transfer ~ Agent",
    data=goals_escapees_actors
    ).fit()

print(RT1_A.summary())

Optimization terminated successfully.
         Current function value: 0.669884
         Iterations 4
                           Logit Regression Results                           
Dep. Variable:      resolved_transfer   No. Observations:                   61
Model:                          Logit   Df Residuals:                       59
Method:                           MLE   Df Model:                            1
Date:                Wed, 01 Jul 2026   Pseudo R-squ.:                 0.03337
Time:                        14:12:03   Log-Likelihood:                -40.863
converged:                       True   LL-Null:                       -42.274
Covariance Type:            nonrobust   LLR p-value:                   0.09299
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept     -0.3795      0.360     -1.054      0.292      -1.085       0.326
Agent          0.8720      0.

**RT2_A Model** 

resolved_transfer ~  Goal_Type

In [94]:
RT2_A = smf.logit(
    "resolved_transfer ~ Goal_Type",
    data=goals_escapees_actors
    ).fit()

print(RT2_A.summary())

Optimization terminated successfully.
         Current function value: 0.682123
         Iterations 4
                           Logit Regression Results                           
Dep. Variable:      resolved_transfer   No. Observations:                   61
Model:                          Logit   Df Residuals:                       59
Method:                           MLE   Df Model:                            1
Date:                Wed, 01 Jul 2026   Pseudo R-squ.:                 0.01571
Time:                        14:12:03   Log-Likelihood:                -41.610
converged:                       True   LL-Null:                       -42.274
Covariance Type:            nonrobust   LLR p-value:                    0.2491
                                     coef    std err          z      P>|z|      [0.025      0.975]
--------------------------------------------------------------------------------------------------
Intercept                         -0.2683      0.368     -0.728     

**RT3_A Model**

resolved_transfer ~  Agent + Goal_Type

In [95]:
RT3_A = smf.logit(
    "resolved_transfer ~ Agent + Goal_Type",
    data=goals_escapees_actors
    ).fit()

print(RT3_A.summary())

Optimization terminated successfully.
         Current function value: 0.659032
         Iterations 4
                           Logit Regression Results                           
Dep. Variable:      resolved_transfer   No. Observations:                   61
Model:                          Logit   Df Residuals:                       58
Method:                           MLE   Df Model:                            2
Date:                Wed, 01 Jul 2026   Pseudo R-squ.:                 0.04903
Time:                        14:12:03   Log-Likelihood:                -40.201
converged:                       True   LL-Null:                       -42.274
Covariance Type:            nonrobust   LLR p-value:                    0.1258
                                     coef    std err          z      P>|z|      [0.025      0.975]
--------------------------------------------------------------------------------------------------
Intercept                         -0.6917      0.460     -1.505     

**RT4_A Model**

resolved_transfer ~ Agent * Goal_Type

In [96]:
RT4_A = smf.logit(
    "resolved_transfer ~ Agent * Goal_Type",
    data= goals_escapees_actors
    ).fit()

print(RT4_A.summary())

Optimization terminated successfully.
         Current function value: 0.654964
         Iterations 5
                           Logit Regression Results                           
Dep. Variable:      resolved_transfer   No. Observations:                   61
Model:                          Logit   Df Residuals:                       57
Method:                           MLE   Df Model:                            3
Date:                Wed, 01 Jul 2026   Pseudo R-squ.:                 0.05490
Time:                        14:12:03   Log-Likelihood:                -39.953
converged:                       True   LL-Null:                       -42.274
Covariance Type:            nonrobust   LLR p-value:                    0.2000
                                           coef    std err          z      P>|z|      [0.025      0.975]
--------------------------------------------------------------------------------------------------------
Intercept                               -0.5108      0.5

In [97]:
RT5_A = smf.logit(
    "resolved_transfer ~ Agent",
    data= no_goal_escapees_actors
    ).fit()

print(RT5_A.summary())

Optimization terminated successfully.
         Current function value: 0.659420
         Iterations 4
                           Logit Regression Results                           
Dep. Variable:      resolved_transfer   No. Observations:                   35
Model:                          Logit   Df Residuals:                       33
Method:                           MLE   Df Model:                            1
Date:                Wed, 01 Jul 2026   Pseudo R-squ.:               0.0004425
Time:                        14:12:03   Log-Likelihood:                -23.080
converged:                       True   LL-Null:                       -23.090
Covariance Type:            nonrobust   LLR p-value:                    0.8863
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      0.4855      0.449      1.080      0.280      -0.395       1.366
Agent          0.1023      0.

4. Model comparisons

In [98]:
lr_test(RT1_A, RT3_A)

LR         1.323856
df         1.000000
p_value    0.249901
dtype: float64

In [99]:
lr_test(RT1_A, RT4_A)

LR         1.820158
df         2.000000
p_value    0.402492
dtype: float64

5. Summary Table

correct_summary_actors

model_name = ["RC1_A", "RC2_A", "RC2_A", "RC3_A", "RC3_A", "RC3_A", "RC4_A", "RC4_A", "RC4_A", "RC4_A", "RC4_A"]

outcome = ["resolved_correct", "resolved_correct", "resolved_correct", "resolved_correct", "resolved_correct", "resolved_correct", "resolved_correct", "resolved_correct", "resolved_correct", "resolved_correct", "resolved_correct"]

subgroup = ["Actors", "Actors", "Actors", "Actors", "Actors", "Actors", "Actors", "Actors", "Actors", "Actors", "Actors"]

formula = ["resolved_correct ~ Agent", "resolved_correct ~ Goal_Type", "resolved_correct ~ Goal_Type", "resolved_correct ~ Agent + Goal_Type", "resolved_correct ~ Agent + Goal_Type", "resolved_correct ~ Agent + Goal_Type", "resolved_correct ~ Agent * Goal_Type", "resolved_correct ~ Agent * Goal_Type", "resolved_correct ~ Agent * Goal_Type", "resolved_correct ~ Agent * Goal_Type", "resolved_correct ~ Agent * Goal_Type"]

predictor= ["Agent", "goal_non_frequent", "no_goal", "goal_non_frequent", "no_goal", "Agent",  "goal_non_frequent", "no_goal", "Agent", "Agent:goal_non_frequent", "Agent:no_goal"]

coef = [1.1259, 0.1382, 0.8473, 0.1448, 0.8790, 1.1455, -3.881e-16, 1.2528, 1.2528, 0.7621, -1.2528]

coef_p = [0.038, 0.793, 0.159, 0.788, 0.151, 0.036, 1.000, 0.097, 0.150, 0.591, 0.338]

LLR_p = [0.02657, 0.3143, 0.3143, 0.06234, 0.06234, 0.06234, 0.09726, 0.09726, 0.09726, 0.09726, 0.09726]

N = [120, 120, 120, 120, 120, 120, 120, 120, 120, 120, 120]

Takeaway = []

correct_summary_actors = pd. DataFrame({
    "Model": model_name,
    "Outcome": outcome,
    "Subgroup": subgroup,
    "Formula": formula, 
    "Predictor": predictor,
    "Coef": coef,
    "Coef_p": coef_p,
    "Model_LLR_p": LLR_p,
    "N": N
})

correct_summary_actors